modules/api/src/main/java/org/apache/ignite/marshaling/ByteArrayMarshaler.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.marshaling;
+
+import java.io.Serializable;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * The ByteArrayMarshaler interface is designed to marshal and unmarshal objects to and from byte arrays. If the byte[] is your preferred
+ * way of marshaling objects, you can implement this interface to provide custom marshaling logic.
+ *
+ * <p>The default implementation of the marshal method is plain java serialization.
+ */
+public interface ByteArrayMarshaler<T> extends Marshaler<T, byte[]> {
+    @Override
+    default byte @Nullable [] marshal(@Nullable T object) {
+        if (object == null) {
+            return null;
+        }
+
+        if (object instanceof Serializable) {
+            return JavaSerializationByteArrayMarshallilng.marshal((Serializable) object);
```

Comment:
Looks like this marshaller is the default, and it does not work with simple types (e.g. String), so it can't be used from non-Java clients. Is that correct?